projects/solutions/zihe/NationalParks/src/components/FilterByState.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { Link } from 'react-router-dom'
+import styled from 'styled-components'
+import { setClickParkName, setDetails, setIsLoading } from '../redux/parkSlice';
+
+const StyledLink = styled(Link)`
+list-style-type: none;
+text-decoration: none;
+color:black;
+width: 130px;
+height:150px;
+margin:5px;
+font-size: 1px;
+overflow:hidden;
+:hover {cursor:pointer;font-weight:600}
+`
+const StyledImg = styled.img`
+  width: 100%;
+  height:100%;
+`
+
+export default function FilterByState() {
+    const { url, allParks } = useSelector((state) => state.park)
+    const [parksByState, setParksByState] = useState([]);
+    const [checkedStates, setCheckedStates] = useState([]);
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        const newState = [];
+        allParks.forEach((park) => {
+
+            const states = park.states.split(',');
+            states.forEach((state) => {
+                if (!newState[state]) {
+                    newState[state] = [];
+                }
+                newState[state].push(
+                    {
+                        id: park.id,
+                        name: park.name,
+                        img: park.img
+                    });
+            });
+        });
+        setParksByState(newState);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    const checkedBoxChange = (e) => {
+        const { name, checked } = e.target;
+        setCheckedStates((p) => ({ ...p, [name]: checked, }))
+    }
+    const clickLink = (id) => {
```

Comment:
should not fetch data for park again.

use navigate to parkDetail page

Reply:
- backend filtering:  if API support, load only those filtered park data
- frontend filtering: do not trim your data -> display all information

Reply:
the same applies for search